pandas/tests/frame/methods/test_replace.py

```diff
@@ -1517,9 +1517,6 @@ def test_replace_with_duplicate_columns(self, replacement):
 
         tm.assert_frame_equal(result, expected)
 
-    @pytest.mark.xfail(
-        reason="replace() changes dtype from period to object, see GH34871", strict=True
-    )
     def test_replace_period_ignore_float(self):
```

Comment:
does this also affect/fix Series.replace?  if so let's make sure we have a test for that too